pkgs/by-name/sw/sword/package.nix

```diff
@@ -33,18 +57,34 @@ stdenv.mkDerivation (finalAttrs: {
     patchShebangs .;
   '';
 
+  # Linux and others will build this way
   configureFlags = [
     "--without-conf"
     "--enable-tests=no"
   ];
 
-  CXXFLAGS = [
-    "-Wno-unused-but-set-variable"
-    "-Wno-unknown-warning-option"
-    # compat with icu61+ https://github.com/unicode-org/icu/blob/release-64-2/icu4c/readme.html#L554
-    "-DU_USING_ICU_NAMESPACE=1"
+  # Windows will build this way. It's slightly less supported, but the configure
+  # script doesn't allow the final linkage to happen with autotools
+  cmakeFlags = lib.optionals isWindows [
+    "-DLIBSWORD_LIBRARY_TYPE=Shared"
+    "-DCROSS_COMPILE_MINGW32=TRUE"
+    "-DICU_CONFIG_OPTS=\"--no-verify\""
   ];
 
+  CXXFLAGS =
```

Comment:
```suggestion
  env.CXXFLAGS =
```

Reply:
`error: The `env` attribute set can only contain derivation, string, boolean or integer attributes. The `CXXFLAGS` attribute is of type list.`

Reply:
Then convert it to string.

Reply:
Is moving this under `env` a new requirement? It seems to be in the minority throughout the codebase.

Reply:
> Is moving this under env a new requirement?

It should.

`CXXFLAGS` is not an attribute that modifies the behavior of  `mkDerivation`.
It is an environment variable that modifies the underlying behavior of GCC.

In order to separate the subjects, we put such environment variables under `env`.

>  It seems to be in the minority throughout the codebase

The code base needs some serious refactoring...

Reply:
I suppose you could use it like: 

```nix
    lib.concatStringsSep " " [
      "-Wno-unused-but-set-variable"
      "-Wno-unknown-warning-option"
      # compat with icu61+ https://github.com/unicode-org/icu/blob/release-64-2/icu4c/readme.html#L554
      "-DU_USING_ICU_NAMESPACE=1"
    ];
```

Reply:
That's exactly what I was going with.